projects/igniteui-angular/src/lib/tabs/tabs-group.component.ts

```diff
@@ -23,46 +23,125 @@ import { IgxTabItemTemplateDirective } from './tabs.directives';
 })
 
 export class IgxTabsGroupComponent implements AfterContentInit, AfterViewChecked {
+    private _itemStyle = 'igx-tabs-group';
+    /**
+     * @hidden
+     */
     public isSelected = false;
 
+    /**
+    * An @Input property that sets the value of the `label`.
+    *```typescript
+    *<igx-tabs-group label="Tab 1" icon="folder">
+    *```
+    */
     @Input()
     public label: string;
 
+    /**
+    * An @Input property that sets the value of the `icon`.
+    *```html
+    *<igx-tabs-group label="Tab 1" icon="home">
+    *```
+    */
     @Input()
     public icon: string;
 
+    /**
+    * An @Input property that allows you to enable/disable the `IgxTabGroupComponent`.
+    *```html
+    *<igx-tabs-group label="Tab 2  Lorem ipsum dolor sit" icon="home" [isDisabled]="true">
+    *```
+    */
     @Input()
     public disabled: boolean;
 
-    @HostBinding('attr.role') public role = 'tabpanel';
-
+    /**
+     * @hidden
+     */
+    @HostBinding('attr.role')
+    public role = 'tabpanel';
+
+    /**
+     * An accessor that returns the class of the `IgxTabsGroupComponent`.
+     * ```typescript
+     * @ViewChild("MyTabsGroup")
+     * public tab: IgxTabsGroupComponent;
+     * ngAfterViewInIt(){
+     *    let tabStyleClass = this.tab.styleClass;
+     * }
+     * ```
+     */
     @HostBinding('class')
     get styleClass(): string {
         return 'igx-tabs__group';
     }
 
+    /**
+     * An accessor that returns the value of the `_itemStyle` property.
```

Comment:
@alexalex972 Maybe this should be hidden or if it's not you should not mention private variables. Internal implementation is a subject of change.